support-frontend/assets/pages/paper-subscription-checkout/components/summary.jsx

```diff
@@ -0,0 +1,230 @@
+// @flow
+
+import React, { Component } from 'react';
+import { hasDiscount, type Price, type Promotion } from 'helpers/productPrice/productPrices';
+import type { BillingPeriod } from 'helpers/billingPeriods';
+import styles from './summary.module.scss';
+import { PriceLabel } from 'components/priceLabel/priceLabel';
+import typeof GridImageType from 'components/gridImage/gridImage';
+import { type GridImg } from 'components/gridImage/gridImage';
+import SvgDropdownArrowUp from './dropDownArrowUp.svg';
+
+// Types
+
+export type DataListItem = {
+  title: string,
+  value: string,
+}
+
+type PropTypes = {|
+  billingPeriod: BillingPeriod,
+  changeSubscription: string,
+  dataList: DataListItem[],
+  description: ?string,
+  image: $Call<GridImageType, GridImg>,
+  productPrice: Price,
+  promotion: ?Promotion,
+  title: string,
+|};
+
+type StateTypes = {
+  showDropDown: boolean,
+}
+
+// Helpers
+
+const DataList = (props: { dataList: DataListItem[] }) => (
+  <div className={styles.dataList}>
+    {props.dataList.length > 0 &&
+      <dl className={styles.data}>
+        {props.dataList.map(item => [
+          <dt>{item.title}</dt>,
+          <dd>{item.value}</dd>,
+            ])
+        }
+      </dl>
+    }
+  </div>
+);
+
+
+const PromotionDiscount = (props: { promotion: ?Promotion }) => (
+  <span>
+    {props.promotion && hasDiscount(props.promotion) &&
+      <div className={styles.promo}>
+        <strong className={styles.promoTitle}>
+          {props.promotion.description}
+        </strong>
+        {' '}({props.promotion.promoCode})
+      </div>
+    }
+  </span>
+);
+
+
+const ChangeSubscription = (props: { route: string }) => (
+  <a className={styles.changeSub} href={props.route}>Change Subscription</a>
+);
+
+
+const DropDownButton = (props: { onClick: Function, showDropDown: boolean }) => (
+  <button
+    aria-hidden="true"
+    className={styles.dropDown}
+    onClick={props.onClick}
+  >
+    <span className={styles.spaceRight}>{props.showDropDown ? 'Hide details' : 'Show all details'}</span>
+    <SvgDropdownArrowUp className={props.showDropDown ? styles.openState : styles.defaultState} />
+  </button>
+);
+
+
+const TabletAndDesktop = (props: {
```

Comment:
Is this type the same as `PropTypes` defined above? If it is you could use that rather than redefining it here.

Reply:
Great point - I love removing code! Thanks